opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/OpenDcsResource.java

```diff
@@ -0,0 +1,47 @@
+/*
+ *  Copyright 2024 OpenDCS Consortium and its Contributors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License")
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.opendcs.odcsapi.res;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+
+class OpenDcsResource
+{
+	private final OpenDcsDatabase db;
+
+	protected OpenDcsResource()
+	{
+		try
+		{
+			//TODO - need to iron out the correct way to handle data source.
+			Context initialCtx = new InitialContext();
```

Comment:
https://github.com/USACE/cwms-data-api/blob/5fd3d7e6d0ead40992bc47631916245575b4d017/cwms-data-api/src/main/java/cwms/cda/ApiServlet.java#L262

This is a "servlet" thing, not a Tomcat/javalin thing. it should generally be injected on each request. Using a base class maybe still work but the DataSource becomes a class property and the getDb would actually create the database.

Debatable if that should, or even can, be a static shared variable. OpenDcsDatabase *should* be thread safe.

Reply:
yeah that's where I was heading. Wasn't sure if we should inject the OpenDcsDatabase as a singleton/shared variable if it's thread safe which would mean we don't want the DataSource itself injected. For the moment I put OpenDcsDatabase in the constructor rather than the getter because it isn't the Closeable instance and so the lifecycle doesn't need to be managed.